pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,290 @@
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBSpring[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	return envProps, nil
+}
+
+func (i *mongoDBInfraReconciler) getMongoDBAppProps(mongoDBInstance *mongodb.MongoDB) (map[string]string, error) {
+	appProps := map[string]string{}
+
+	mongoDBURI := mongoDBInstance.Status.MongoURI
+	if len(mongoDBURI) > 0 {
+		mongoDBParsedURL, err := url.ParseRequestURI(mongoDBURI)
+		if err != nil {
+			return nil, err
+		}
+		appProps[propertiesMongoDBSpring[appPropMongoDBHost]] = mongoDBParsedURL.Hostname()
+		appProps[propertiesMongoDBSpring[appPropMongoDBPort]] = mongoDBParsedURL.Port()
+		appProps[propertiesMongoDBQuarkus[appPropMongoDBURI]] = mongoDBURI
+	}
+
+	return appProps, nil
+}
+
+func (i *mongoDBInfraReconciler) updateMongoDBAppPropsInStatus(mongoDBInstance *mongodb.MongoDB) error {
+	log.Debugf("going to Update MongoDB app properties in kogito infra instance status")
+	appProps, err := i.getMongoDBAppProps(mongoDBInstance)
+	if err != nil {
+		return err
+	}
+	i.instance.Status.AppProps = appProps
+	log.Debugf("Following app properties are set infra status : %s", appProps)
+	return nil
+}
+
+func (i *mongoDBInfraReconciler) updateMongoDBEnvVarsInStatus(mongoDBInstance *mongodb.MongoDB) error {
+	log.Debugf("going to Update MongoDB env properties in kogito infra instance status")
+	envVars, err := i.getMongoDBSecretEnvVars(mongoDBInstance)
+	if err != nil {
+		return err
+	}
+	i.instance.Status.Env = envVars
+	log.Debugf("Following env properties are set infra status : %s", envVars)
+	return nil
+}
+
+func (i *mongoDBInfraReconciler) loadDeployedMongoDBInstance(instanceName string, namespace string) (*mongodb.MongoDB, error) {
+	log.Debug("fetching deployed kogito mongoDB instance")
+	mongoDBInstance := &mongodb.MongoDB{}
+	if exits, err := kubernetes.ResourceC(i.client).FetchWithKey(types.NamespacedName{Name: instanceName, Namespace: namespace}, mongoDBInstance); err != nil {
+		log.Error("Error occurs while fetching kogito mongoDB instance")
+		return nil, err
+	} else if !exits {
+		log.Debug("Kogito mongoDB instance is not exists")
+		return nil, nil
+	} else {
+		log.Debug("Kogito mongoDB instance found")
+		return mongoDBInstance, nil
+	}
+}
+
+func (i *mongoDBInfraReconciler) loadCustomKogitoMongoDBSecret(namespace string) (*corev1.Secret, error) {
+	log.Debugf("Fetching %s ", mongoDBSecretName)
+	secret := &corev1.Secret{}
+	if exits, err := kubernetes.ResourceC(i.client).FetchWithKey(types.NamespacedName{Name: mongoDBSecretName, Namespace: namespace}, secret); err != nil {
+		log.Errorf("Error occurs while fetching %s", mongoDBSecretName)
+		return nil, err
+	} else if !exits {
+		log.Errorf("%s not found", mongoDBSecretName)
```

Comment:
We might need to return a specialized error here to reconcile after T? Take a look at the `errors.go` file.

Reply:
I think this needs more redo on all the infra controllers.
I have just did the think similar to https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/controller/kogitoinfra/infinispan.go#L288

Reply:
Sure! But can't we start here and open a JIRA to follow up others? Optional, do if you have the time. Otherwise a follow up JIRA would do.

Reply:
Coming back to that, why would we need to reconcile ?
this is something that the operator is creating for the application to have auth information...
we could when creating it though

Reply:
thinking of something. testing it and then get back ;)

Reply:
added https://github.com/kiegroup/kogito-cloud-operator/pull/627/files#diff-65b99f6143ba7d89e73a9c85839d59c54bcf857ef2a77919de76c68165d3e9dcR63-R69